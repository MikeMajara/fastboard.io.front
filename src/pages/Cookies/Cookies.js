import React from 'react';
import { useTranslation } from 'react-i18next';

import { HeadMeta } from '#components';
import routes from '#routes';

import {
  SWrapper, SHeader, SSubHeader, SContent, SList, SListItem, SListItemName,
} from './styled';

const Cookies = () => {
  const { t } = useTranslation('cookies');

  return (
    <>
      <HeadMeta route={routes.cookies} />
      <SWrapper>
        <SHeader>
          {t('title')}
        </SHeader>
        <SContent>
          {t('subtitle')}
        </SContent>

        <SSubHeader>
          {t('what.heading')}
        </SSubHeader>
        <SContent>
          {t('what.p1')}
        </SContent>
        <SContent>
          {t('what.p2')}
        </SContent>

        <SSubHeader>
          {t('which.heading')}
        </SSubHeader>
        <SContent>
          {t('which.p1')}
        </SContent>
        <SContent>
          {t('which.p2.heading')}
          <SList>
            <SListItem>
              {t('which.p2.list.own.description')}
              <SList>
                <SListItem>
                  <SListItemName>
                    {t('which.p2.list.own.cookies.language.name')}
                    :
                  </SListItemName>
                  {t('which.p2.list.own.cookies.language.description')}
                </SListItem>
                <SListItem>
                  <SListItemName>
                    {t('which.p2.list.own.cookies.allow.name')}
                    :
                  </SListItemName>
                  {t('which.p2.list.own.cookies.allow.description')}
                </SListItem>
              </SList>
            </SListItem>
            <SListItem>
              {t('which.p2.list.thirdParty.description')}
              <SList>
                <SListItem>
                  <SListItemName>
                    {t('which.p2.list.thirdParty.cookies.analytics.name')}
                    :
                  </SListItemName>
                  {t('which.p2.list.thirdParty.cookies.analytics.description')}
                </SListItem>
              </SList>
            </SListItem>
          </SList>
        </SContent>

        <SSubHeader>
          {t('acceptance.heading')}
        </SSubHeader>
        <SContent>
          {t('acceptance.p1')}
        </SContent>
        <SContent>
          {t('acceptance.p2')}
        </SContent>
        <SContent>
          {t('acceptance.p3')}
        </SContent>
      </SWrapper>
    </>
  );
};

export default Cookies;
