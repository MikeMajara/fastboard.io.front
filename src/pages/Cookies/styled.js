import styled, { css, breakpoints } from '@xstyled/styled-components';

export const SWrapper = styled.div`
  margin: auto;
  text-align: justify;
  max-width: containerMaxWidth;

  ${breakpoints({
    xs : css`padding: default;`,
    sm : css`padding: 40px;`,
    md : css`padding: 40px 80px;`,
    lg : css`padding: 40px 120px;`,
    xl : css`padding: 40px 200px;`,
  })}
`;

export const SHeader = styled.h1`
  margin-top: default;
  margin-bottom: default;
`;

export const SSubHeader = styled.h2`
  margin-top: default;
  margin-bottom: default;
`;

export const SContent = styled.div`
  margin-top: default;
`;

export const SList = styled.ul`
`;

export const SListItem = styled.li`
  margin-top: sm;
`;

export const SListItemName = styled.span`
  margin-right: xs;
  font-weight: bold;
`;
