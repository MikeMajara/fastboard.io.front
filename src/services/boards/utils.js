/**
 * Send boards event trough the network.
 *
 * @param {String} eventName Event key name.
 * @param {Object} data Data associated to the event.
 * @param {Function} ask Callback for server event acknowledgement.
 */
function send(eventName, data, ack) {
  this.socket.emit(eventName, data, ack);
}

/**
 * Check if the service is initialized.
 *
 * @return {Boolean} True if the service is initialized; false otherwise.
 */
function isInit() {
  return this.isInit;
}

export {
  isInit,
  send,
};
