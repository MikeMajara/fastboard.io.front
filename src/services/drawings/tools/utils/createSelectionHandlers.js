import { Group, Shape } from 'paper';

import { Log } from '#utils';
import { cursorTypes, selectionColorCode } from '#constants';
import store, { setSelectorCursorHover } from '#store';

import { bounds } from '../selector/constants';

const boundsArray = Object.values(bounds);
const bounds2cursorType = Object.freeze({
  [bounds.topLeft]     : cursorTypes.rotating,
  [bounds.topRight]    : cursorTypes.menu,
  [bounds.bottomLeft]  : cursorTypes.resizingSW,
  [bounds.bottomRight] : cursorTypes.resizingSE,
});

/**
 * Add selection handlers to the selected item.
 *
 * @param {Object} selectedItem Item to be added selection handlers to.
 * @param {Object} selectionLayer Layer that will contain the selection handlers.
 *
 * @returns {Object} Item - group with 4 handler items - from the selection handler layer.
 */
export default function createSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : createSelectionHandlers', { selectedItem });

  // Set selected item to show its selection state
  selectedItem.set({
    selected      : true,
    selectedColor : selectionColorCode,
    onMouseEnter  : () => store.dispatch(setSelectorCursorHover(cursorTypes.dragging)),
    onMouseLeave  : () => store.dispatch(setSelectorCursorHover()),
  });

  // Create bound handlers on the selection layer
  boundsArray.forEach((bound) => {
    selectionLayer.addChild(new Shape.Circle({
      name         : bound,
      center       : selectedItem.internalBounds[bound],
      radius       : 6,
      fillColor    : selectionColorCode,
      onMouseEnter : () => store.dispatch(setSelectorCursorHover(bounds2cursorType[bound])),
      onMouseLeave : () => store.dispatch(setSelectorCursorHover()),
    }));
  });

  // Create a group with the handlers to manipulate all of them at once
  selectionLayer.addChild(
    new Group({
      name     : 'selectionHandlers', // @todo to constants
      children : boundsArray.map(
        (bound) => selectionLayer.children[bound],
      ),
    }),
  );

  // Reproduce the selected item transformations on the handler items group
  selectionLayer.children.selectionHandlers.transform(selectedItem.matrix);

  // Except the scaling of the handler items themselves, which retain the original size
  selectionLayer.children.selectionHandlers.children.forEach((handlerItem) => {
    handlerItem.set({ scaling : 1 });
  });

  return selectionLayer.children.selectionHandlers;
}
