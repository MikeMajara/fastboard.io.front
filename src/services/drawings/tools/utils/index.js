export { default as createSelectionHandlers } from './createSelectionHandlers';
export { default as removeSelectionHandlers } from './removeSelectionHandlers';
export { default as resizeSelectionHandlers } from './resizeSelectionHandlers';
export { default as rotateSelectionHandlers } from './rotateSelectionHandlers';
export { default as translateSelectionHandlers } from './translateSelectionHandlers';
