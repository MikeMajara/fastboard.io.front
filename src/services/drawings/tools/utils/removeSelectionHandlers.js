import { Log } from '#utils';

/**
 * Remove selection handlers.
 *
 * @param {Object} selectedItem Item to be removed selection handlers from.
 * @param {Object} selectionLayer Layer that contains the selection handlers.
 */
export default function removeSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : removeSelectionHandlers');

  selectedItem.set({ selected : false });
  selectionLayer.removeChildren();
}
