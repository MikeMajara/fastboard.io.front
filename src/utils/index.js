export { default as Log } from './logger';
export { default as generateId } from './generateId';
export { default as point2net } from './point2net';
export { default as setPreventUnload } from './setPreventUnload';
export { default as throttle } from './throttle';
export { default as timeoutPromise } from './timeoutPromise';
export { default as triggerDownload } from './triggerDownload';
export { default as triggerUpload } from './triggerUpload';
export { default as validBoardId } from './validBoardId';
