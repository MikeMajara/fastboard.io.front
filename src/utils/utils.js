/**
 * Throttles a function call.
 *
 * @param {Function} callback Function to be throttled.
 * @param {Integer} delay Throttle delay in milliseconds.
 *
 * @return {Function} Function that throttles the execution of the callback.
 */
export const throttle = (callback, delay = 0) => {
  let previousTime = new Date().getTime();

  return function throttledCallback(args) {
    const currentTime = new Date().getTime();

    if ((currentTime - previousTime) >= delay) {
      previousTime = currentTime;
      callback(...args);
    }
  };
};

/**
 * Creates a Promise with a rejection timeout.
 *
 * @param {Function} callback Function for the promise. Receives res and rej functions.
 * @param {Integer} timeout Timeout for the promise to be rejected in milliseconds.
 *
 * @return {Promise} Promise with rejection timeout.
 */
export const timeoutPromise = (callback, timeout) => new Promise(
  (res, rej) => {
    const timeoutId = setTimeout(() => rej(new Error()), timeout);

    callback(
      (...args) => {
        clearTimeout(timeoutId);
        res(...args);
      },
      (...args) => {
        clearTimeout(timeoutId);
        rej(new Error(...args));
      },
    );
  },
);
