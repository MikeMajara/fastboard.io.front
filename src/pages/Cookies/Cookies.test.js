import React from 'react';
import { HelmetProvider } from 'react-helmet-async';

import { render } from '#devTools/tests/utils';

import Cookies from './Cookies';

jest.mock('#services', () => ({
  analyticsService : {
    pageview : jest.fn(),
  },
}));

describe('Page : Cookies', () => {
  test('Snapshot', () => {
    const { asFragment } = render(
      <HelmetProvider>
        <Cookies />
      </HelmetProvider>,
      { router : true },
    );

    const componentRender = asFragment();
    expect(componentRender).toMatchSnapshot();
  });
});
