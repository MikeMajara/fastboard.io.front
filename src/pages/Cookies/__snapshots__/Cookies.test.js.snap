// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page : Cookies Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-prQdK HZjFt"
  >
    <h1
      class="sc-qQlgh dSJdA"
    >
      title
    </h1>
    <div
      class="sc-pAYXY imdwhw"
    >
      subtitle
    </div>
    <h2
      class="sc-qYhTA dPgfEO"
    >
      what.heading
    </h2>
    <div
      class="sc-pAYXY imdwhw"
    >
      what.p1
    </div>
    <div
      class="sc-pAYXY imdwhw"
    >
      what.p2
    </div>
    <h2
      class="sc-qYhTA dPgfEO"
    >
      which.heading
    </h2>
    <div
      class="sc-pAYXY imdwhw"
    >
      which.p1
    </div>
    <div
      class="sc-pAYXY imdwhw"
    >
      which.p2.heading
      <ul
        class="sc-pJwpB eqQGcv"
      >
        <li
          class="sc-pRSgm fwojeL"
        >
          which.p2.list.own.description
          <ul
            class="sc-pJwpB eqQGcv"
          >
            <li
              class="sc-pRSgm fwojeL"
            >
              <span
                class="sc-qapxP gJoMDM"
              >
                which.p2.list.own.cookies.language.name:
              </span>
              which.p2.list.own.cookies.language.description
            </li>
            <li
              class="sc-pRSgm fwojeL"
            >
              <span
                class="sc-qapxP gJoMDM"
              >
                which.p2.list.own.cookies.allow.name:
              </span>
              which.p2.list.own.cookies.allow.description
            </li>
          </ul>
        </li>
        <li
          class="sc-pRSgm fwojeL"
        >
          which.p2.list.thirdParty.description
          <ul
            class="sc-pJwpB eqQGcv"
          >
            <li
              class="sc-pRSgm fwojeL"
            >
              <span
                class="sc-qapxP gJoMDM"
              >
                which.p2.list.thirdParty.cookies.analytics.name:
              </span>
              which.p2.list.thirdParty.cookies.analytics.description
            </li>
            <li
              class="sc-pRSgm fwojeL"
            >
              <span
                class="sc-qapxP gJoMDM"
              >
                which.p2.list.thirdParty.cookies.cloudflare.name:
              </span>
              which.p2.list.thirdParty.cookies.cloudflare.description
            </li>
          </ul>
        </li>
      </ul>
    </div>
    <h2
      class="sc-qYhTA dPgfEO"
    >
      acceptance.heading
    </h2>
    <div
      class="sc-pAYXY imdwhw"
    >
      acceptance.p1
    </div>
    <div
      class="sc-pAYXY imdwhw"
    >
      acceptance.p2
    </div>
    <div
      class="sc-pAYXY imdwhw"
    >
      acceptance.p3
    </div>
  </div>
</DocumentFragment>
`;
